fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1157,6 +1187,167 @@ long getTotalRecordsScanned() {
         return totalRecordsScanned.get();
     }
 
+    /**
+     * A class holder for the config parameters needed to rebuild an online index.
+     */
+    @API(API.Status.UNSTABLE)
+    public static class Config {
+        private final int maxLimit;
+        private final int maxRetries;
+        private final int recordsPerSecond;
+        private final long progressLogIntervalMillis;
+        private final int increaseLimitAfter;
+
+        public Config(int maxLimit, int maxRetries, int recordsPerSecond, long progressLogIntervalMillis, int increaseLimitAfter) {
+            this.maxLimit = maxLimit;
+            this.maxRetries = maxRetries;
+            this.recordsPerSecond = recordsPerSecond;
+            this.progressLogIntervalMillis = progressLogIntervalMillis;
+            this.increaseLimitAfter = increaseLimitAfter;
+        }
+
+        /**
+         * Get the maximum number of records to process in one transaction.
+         * @return the maximum number of records to process in one transaction
+         */
+        public int getMaxLimit() {
+            return maxLimit;
+        }
+
+        /**
+         * Get the maximum number of times to retry a single range rebuild.
+         * @return the maximum number of times to retry a single range rebuild
+         */
+        public int getMaxRetries() {
+            return maxRetries;
+        }
+
+        /**
+         * Get the maximum number of records to process in a single second.
+         * @return the maximum number of records to process in a single second
+         */
+        public int getRecordsPerSecond() {
+            return recordsPerSecond;
+        }
+
+        /**
+         * Get the minimum time between successful progress logs when building across transactions.
+         * Negative will not log at all, 0 will log after every commit.
+         * @return the minimum time between successful progress logs in milliseconds
+         */
+        public long getProgressLogIntervalMillis() {
+            return progressLogIntervalMillis;
+        }
+
+        /**
+         * Get the number of successful range builds before re-increasing the number of records to process in a single
+         * transaction.
+         * By default this is {@link #DO_NOT_RE_INCREASE_LIMIT}, which means it will not re-increase after successes.
+         * @return the number of successful range builds before increasing the number of records processed in a single
+         * transaction
+         */
+        public int getIncreaseLimitAfter() {
+            return increaseLimitAfter;
+        }
+
+        @Nonnull
+        public static Builder newBuilder() {
+            return new Builder();
+        }
+
+        /**
+         * A builder for {@link Config}. These are the configuration parameters used while rebuilding indexes.
```

Comment:
```suggestion
         * A builder for {@link Config}. These are the configuration parameters used while building indexes.
```

And a similar comment as above about the term "configuration parameters" here.

Reply:
done.